models/baseModels/Invoice/Invoice.ts

```diff
@@ -189,13 +197,40 @@ export abstract class Invoice extends Transactional {
     }
 
     await this._updateIsItemsReturned();
+    if (!this.loyaltyProgram) {
+      return;
+    }
+    const doc = (await this.fyo.doc.getDoc(
```

Comment:
It would be better if you name the `doc` variable to a relatable name to its value. like `loyaltyProgram` or `loyaltyProgramDoc`